### ISSUE TITLE
vim-slime doesn't work with kitty

### ISSUE THREAD

**Issue by Svzer** [jpalardy/vim-slime#279]
Hi !

I've recently started to use vim-slime with tmux and it worked flawlessly, but with tmux being unable to display images in the terminal, I wanted to try and rely solely on kitty. However, I just can't seem to make it work : every time I enter the target window id, nothing happens. It does seem like the content is being passed to the .slime_paste file, but it is never sent to the target window.

I would gladly appreciate some help. Thanks in advance !

**Comment by jpalardy** [comment]
Hi @Svzer 

I've also seen weird behavior while trying to display images in iterm2, using tmux. (sometimes it works, sometimes it doesn't … something it doesn't leave enough space for the whole height)

The kitty integration is super simple:

![Screen Shot 2021-01-06 at 09 16 38](https://user-images.githubusercontent.com/18142/103799429-ee684e00-4fff-11eb-890a-e5becec2de20.png)

Can you try to:
- create a file and put content in it (or re-use the slime_paste file)
- manually run the command from line 49 manually and see what happens

Let me know how that goes and we'll take it from there.

**Comment by Svzer** [comment]
Hi,

I'm sorry I forgot to mention this : I already tried manually running this command, and it worked.
I'm very much a novice, but is there a need for the slime_paste file for the kitty integration though ? kitty is able to read text sent from stdin, so I thought perhaps this could be a workaround.

**Comment by jpalardy** [comment]
Hi @Svzer 

I installed kitty and … it doesn't seem to work for me (even manually)

Can you give me a step-by-step?
- which version you're running
- what's in your kitty.conf (saw that allow_remote_control might be important)
- what command you tried — and which ones worked?

If I can reproduce what works/doesn't, that would help 😄

**Comment by Svzer** [comment]
Hi,

I'm running the **0.19.3** version.
Here's my kitty configuration : 
```
# Color theme
include snazzy.conf

font_family	MesloLGSDZ Nerd Font
font_size	12
window_margin_width 0 10

# Tabs
tab_bar_style powerline
active_tab_background #5AF78E
active_tab_font_style bold
inactive_tab_font_style normal

# Windows
active_border_color #5AF78E
window_border_width 2px
window_margin_width 2
window_padding_width 4

# Allow remote control for vim-slime to work
allow_remote_control yes
```
What I've done is : 

- Open a new kitty terminal
- Open a new window inside that terminal with Ctrl+Shift+Enter
- Run the command ```kitty @ send-text --match id:2 --from-file ~/.slime_paste``` inside the first window. I tried it with ipython running in the second window and some very simple python code in the slime_paste file and it worked fine.
- The command ```kitty @ send-text --match id:2 --stdin``` also allows me to send input directly to the second window just fine

**Comment by jpalardy** [comment]
Interesting … `kitty @ send-text` works when running from a kitty terminal, but not from "outside" (iterm2, in my case)

if I run vim inside kitty, vim-slime works just fine 😄 

a couple of things:
- where do you run your vim from?
- does vim-slime work for you inside kitty?
- I wonder how kitty knows "send-text" comes from inside/outside kitty? (is there a security thing in play?)

**Comment by jpalardy** [comment]
Here it is:

![Screen Shot 2021-01-06 at 11 38 56](https://user-images.githubusercontent.com/18142/103812728-cedb2080-5013-11eb-889e-515dccbff7ef.png)

from [this page](https://github.com/kovidgoyal/kitty/blob/master/docs/remote-control.rst)

I wonder if this changed 🤔 

@mdzhang any thoughts on this?

**Comment by Svzer** [comment]
Alright, I've figured it out : it didn't make sense that it would work for you and not for me, so the only difference left was that I use neovim and not the good old vim. After quickly configuring it, vim-slime DOES work with kitty ! The issue was actually neovim the entire time.

**Comment by mdzhang** [comment]
Hmm, I currently still use it with regular VIM 8.2, Kitty 0.18.3, and [this kitty conf](https://github.com/mdzhang/dotfiles/blob/master/kitty/.config/kitty/kitty.conf) without issue. I haven't played around with sending it from outside kitty e.g. iTerm, so I'm not surprised that doesn't work. 

If I switch to `nvim`, I notice it does go to the slime file but not to the target window, so it seems to be something about the interplay of neovim and kitty

**Comment by jpalardy** [comment]
@Svzer @mdzhang thanks for clarifying!

At this point, I'm not sure what's next … maybe a documentation patch to clarify how to configure kitty, gotchas, etc? (neovim, inside kitty…)

As for neovim, it's probably a question of how the commands are shelled out. I don't know if this is worth exploring full `--listen-on` and `--to` flags to the vim-slime setup. (as per [this page](https://sw.kovidgoyal.net/kitty/remote-control.html))

**Comment by mdzhang** [comment]
@Svzer can try and see if using https://github.com/jpalardy/vim-slime/pull/280 works for him

if not, unless a dedicated nvim user wants to dive deeper, a doc patch mentioning the neovim/kitty gotcha sounds good to me

**Comment by Svzer** [comment]
It works perfectly, thank you !

**Comment by jpalardy** [comment]
fixed by #281 

re-open if I missed anything
